src/jobservice/main.go

```diff
@@ -39,6 +42,21 @@ func main() {
 		panic(fmt.Sprintf("failed to load configuration, error: %v", err))
 	}
 
+	// init cache if cache layer enabled
+	// gc needs to delete artifact by artifact manager, but the artifact cache store in
+	// core redis db so here require core redis url and init default cache.
+	if cfgLib.CacheEnabled() {
+		cacheURL := os.Getenv("_REDIS_URL_CORE")
+		u, err := url.Parse(cacheURL)
+		if err != nil {
+			panic("bad _REDIS_URL_CORE")
+		}
+
+		if err = cache.Initialize(u.Scheme, cacheURL); err != nil {
+			panic(fmt.Sprintf("failed to initialize cache: %v", err))
```

Comment:
we use restart to guarantee the timing issues?

Reply:
Yes, same logic with harbor core, we should panic for retry if connect redis error.